let DEBUG_MODE,ValidationError,CustomErrorHandler;_2d4‍.x([["default",()=>_2d4‍.o]]);_2d4‍.w("../config",[["DEBUG_MODE",["DEBUG_MODE"],function(v){DEBUG_MODE=v}]]);_2d4‍.w("joi",[["ValidationError",["ValidationError"],function(v){ValidationError=v}]]);_2d4‍.w("../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);


const errorHandler = (err, req, res, next) => {
  //default err
  let statusCode = 500;

  let data = {
    message: "Internal server error",
    ...(DEBUG_MODE === "true" && { originalError: err.message }),
  };

  //validation err
  if (err instanceof ValidationError) {
    statusCode = 422;
    data = {
      message: err.message,
    };
  }

  //user exists err
  if (err instanceof CustomErrorHandler) {
    statusCode = err.status;
    data = {
      message: err.message,
    };
  }

  return res.status(statusCode).json(data);
};

_2d4‍.d(errorHandler);
